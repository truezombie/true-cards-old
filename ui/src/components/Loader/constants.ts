export const LOADER_SIZE = 30;
export const LOADER_THICKNESS = 5;
