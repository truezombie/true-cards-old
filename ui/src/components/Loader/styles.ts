import { createStyles, Theme } from '@material-ui/core/styles';
import { StyleRules } from '@material-ui/styles/withStyles';

const loaderStyles = (): StyleRules =>
  createStyles({
    root: {
      margin: 'auto',
    },
  });

const linearLoaderStyles = (theme: Theme): StyleRules =>
  createStyles({
    progressWrap: {
      width: '100%',
    },
    progressHeight: {
      height: theme.spacing(1),
    },
  });

export { loaderStyles, linearLoaderStyles };
