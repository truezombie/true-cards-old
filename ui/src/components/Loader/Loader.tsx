import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import { WithStyles, withStyles } from '@material-ui/core/styles';

import { loaderStyles } from './styles';
import { LOADER_SIZE, LOADER_THICKNESS } from './constants';

export type LoaderProps = WithStyles<typeof loaderStyles>;

const Loader = ({ classes }: LoaderProps): JSX.Element => (
  <CircularProgress
    className={classes.root}
    color='primary'
    size={LOADER_SIZE}
    thickness={LOADER_THICKNESS}
  />
);

export default withStyles(loaderStyles)(Loader);
