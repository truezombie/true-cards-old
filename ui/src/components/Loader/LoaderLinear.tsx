import React from 'react';
import Fade from '@material-ui/core/Fade';
import LinearProgress from '@material-ui/core/LinearProgress';
import { WithStyles, withStyles } from '@material-ui/core/styles';
import { linearLoaderStyles } from './styles';

export interface LoaderLinearProps
  extends WithStyles<typeof linearLoaderStyles> {
  show: boolean;
}

const LoaderLinear = ({ classes, show }: LoaderLinearProps): JSX.Element => {
  return (
    <Fade in={show}>
      <div className={classes.progressWrap}>
        <LinearProgress className={classes.progressWrap} />
      </div>
    </Fade>
  );
};

export default withStyles(linearLoaderStyles)(LoaderLinear);
