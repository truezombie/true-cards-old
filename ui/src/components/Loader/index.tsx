import Loader from './Loader';
import LoaderLinear from './LoaderLinear';

export { Loader, LoaderLinear };
