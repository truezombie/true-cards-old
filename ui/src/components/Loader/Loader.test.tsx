import React from 'react';
import { shallow } from 'enzyme';

import Loader, { LoaderProps } from './Loader';

const setUp = (props: LoaderProps) => shallow(<Loader {...props} />);

describe('<Loader />', () => {
  it('should render Loader component', () => {
    const wrapper = setUp({ classes: { root: 'root' } });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
