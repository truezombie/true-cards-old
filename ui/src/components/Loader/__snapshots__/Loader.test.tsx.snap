// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render Loader component 1`] = `"<div class=\\"MuiCircularProgress-root Loader-root-1 root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate\\" style=\\"width:30px;height:30px\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"19.5\\" fill=\\"none\\" stroke-width=\\"5\\"></circle></svg></div>"`;
